src/SpendPermissionManager.sol

```diff
@@ -735,19 +732,19 @@ contract SpendPermissionManager is EIP712 {
             _expectedReceiveAmount = 0;
             // forward native token to recipient, which will revert if funds are not actually available
             SafeTransferLib.safeTransferETH(payable(recipient), value);
-            return;
+        } else {
+            // if ERC-20 token, set allowance for this contract to spend on behalf of account
+            _execute({
+                account: account,
+                target: token,
+                value: 0,
+                data: abi.encodeWithSelector(IERC20.approve.selector, address(this), value)
+            });
+
+            // use ERC-20 allowance to transfer from account to recipient
+            // safeTransferFrom will revert if transfer fails, regardless of ERC-20 implementation
+            IERC20(token).safeTransferFrom(account, recipient, value);
         }
```

Comment:
The early return is leftover from a time where somehow it looked cleaner. Rereading it, I think it's much more straightforward to read as
```
if (native token) {
  // eth logic...
} else {
  // erc20 logic...
}

Reply:
agree